### PR TITLE
feat(CX-3413): Add confidentialNotes field to My Collection artwork form

### PR DESCRIPTION
**Pull request by MrSltun**
The type of this PR is: **Feat**

This PR solves [CX-3413]

### Description
This PR adds the confidential notes field to My Collection artwork form

### Screenshots

| mWeb | Desktop | 
| --- | --- |
| image | image |

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CX-3413]: https://artsyproduct.atlassian.net/browse/CX-3413?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/MyCollection/Routes/EditArtwork/Components/MyCollectionArtworkFormDetails.tsx [hunk 1]
```diff
@@ -325,13 +326,28 @@ export const MyCollectionArtworkFormDetails: React.FC = () => {
             title="City"
             name="artworkLocation"
             placeholder="City where artwork is located"
-            maxLength={256}
+            maxLength={500}
```
Comment (olerichter00) [inline src/Apps/MyCollection/Routes/EditArtwork/Components/MyCollectionArtworkFormDetails.tsx:329]: **question:** Is this change intentional?

**Review by olerichter00 — APPROVED** [review]
Great work 🌟

File: src/Apps/MyCollection/Routes/EditArtwork/Components/MyCollectionArtworkFormDetails.tsx (hunk 1 above)
Comment (olerichter00) [reply src/Apps/MyCollection/Routes/EditArtwork/Components/MyCollectionArtworkFormDetails.tsx:329]: The `provenance` input should be limited to 500 as well ([Figma design](https://www.figma.com/file/99wX65Prz2lTsz9xLm5IrU/Artwork-Detail-Page?node-id=614%3A14139&t=aKmdrtYs9KzSNViV-0)).

Comment (MrSltun) [reply src/Apps/MyCollection/Routes/EditArtwork/Components/MyCollectionArtworkFormDetails.tsx:329]: Yes this was intentional, I believe it is specified in the Figma file

**Review by olerichter00 — DISMISSED** [review]
